agent-tooling/src/main/java/io/opentelemetry/auto/typed/tracer/ClientTypedTracer.java

```diff
@@ -0,0 +1,18 @@
+package io.opentelemetry.auto.typed.tracer;
+
+import io.opentelemetry.auto.typed.span.ClientTypedSpan;
+import io.opentelemetry.trace.Span;
+
+public abstract class ClientTypedTracer<
+        T extends ClientTypedSpan<T, REQUEST, RESPONSE>, REQUEST, RESPONSE>
```

Comment:
Eeek 😆

Reply:
yeah, there wasn't a good way to ensure the return types were strongly typed without some extra generics complexity.